vmaas/rpm_pkg_names.go

```diff
@@ -0,0 +1,52 @@
+package vmaas
+
+import (
+	"slices"
+
+	"github.com/redhatinsights/vmaas-lib/vmaas/utils"
+)
+
+type ContentData map[string][]string
+
+type RPMPkgNames struct {
+	Names      ContentData `json:"rpm_name_list"`
+	LastChange string      `json:"last_change"`
+}
+
+func (c *Cache) getContentData(rpmNames []string, contentSets []string) ContentData {
+	csMap := make(map[string]bool, len(contentSets))
+	for _, cs := range contentSets {
+		csMap[cs] = true
+	}
+
+	isDuplicate := make(map[string]bool, len(rpmNames))
+	contentData := make(ContentData, len(rpmNames))
+	for _, name := range rpmNames {
+		nameID, found := c.Packagename2ID[name]
+		if !found || isDuplicate[name] {
+			continue
+		}
+
+		csIDs := c.nameID2ContentSetIDs(nameID)
+		labels := c.contentSetIDs2Labels(csIDs)
+		if len(contentSets) != 0 {
+			labels = utils.Intersection(labels, csMap)
+		}
+		slices.Sort(labels)
+		contentData[name] = labels
+		isDuplicate[name] = true
+	}
+	return contentData
+}
+
+func (req *RPMPkgNamesRequest) rpmPkgNames(c *Cache) *RPMPkgNames { // TODO: implement opts
+	if len(req.RPMNames) == 0 {
+		return &RPMPkgNames{}
```

Comment:
Will `"'rpm_name_list' is a required property"` be handled in vmaas?
Just to clarify what I meant by the previous comment. If rpm_name_list is in the request but it is [], then the we should respond with empty response. When `rpm_name_list` is missing in the request then we should return error